### DISCUSSION TITLE
deploy to github

### DISCUSSION THREAD

**Discussion by christjohannes** [observablehq/framework#1690] [Q&A] [answered]
I have created an empty observable framework project and deployed it publicly to observable: https://observablehq.com/projects/@rnd/test

After that I linked it to an empty public repository on github and wanted to deploy it to this repository: https://github.com/christjohannes/test

My aim is to update a framework project daily via github actions. 

But the deploying to github throws an error. This is my log: 
```
07:44:03.720Z
dispatch
-
Claimed build job b9f30feff7da2472 from the queue
07:44:03.775Z
dispatch
-
Building project on builder de9ba6cf05043ca8
07:44:03.801Z
prepare
-
Starting build for test
07:44:03.801Z
git
-
Cloning https://github.com/christjohannes/test using ref default branch
07:44:03.802Z
docker
-
Checking for Docker image ghcr.io/observablehq/framework-runtime:latest
07:44:03.806Z
docker
-
Using existing local Docker image
07:44:03.880Z
build
-
Starting build
07:44:03.880Z
git
-
Checked out 9bb06a4b81e200f0d780a79c8de9b79e50083ffc
07:44:03.880Z
prepare
-
No package manager lock file detected, defaulting to npm.Starting container
07:44:04.732Z
dependencies
-
Installing dependencies using npm
07:44:04.998Z
dependencies
-
stderr: npm error path /project/package.json
07:44:04.998Z
dependencies
-
stderr: npm error errno -2
07:44:04.998Z
dependencies
-
stderr: npm error enoent Could not read package.json: Error: ENOENT: no such file or directory, open '/project/package.json'
07:44:04.998Z
dependencies
-
stderr: npm error code ENOENT
07:44:04.998Z
dependencies
-
stderr: npm error enoent This is related to npm not being able to find a file.
07:44:04.998Z
dependencies
-
npm error enoent
07:44:04.998Z
dependencies
-
stderr: npm error syscall open
07:44:04.999Z
dependencies
-
stderr: npm error A complete log of this run can be found in: /home/observable-builder/.npm/_logs/2024-09-27T07_44_04_800Z-debug-0.log
07:44:05.977Z
build
-
Error with build: Command failed exitCode: 254
```

**Comment by mythmon** [comment, marked as answer]
You cannot deploy an empty repository to the Observable Cloud, there is nothing to deploy. You'll need to put Framework code in the repository.

Did you create the data app initially with "npm run deploy"? The place you ran that is the repository you should push to GitHub.

Once you have that working, you won't need need to use GitHub Actions at all, Observable Cloud has a built in scheduling feature. You can see the docs for it here: https://observablehq.com/documentation/data-apps/schedules

**Reply by mbostock** [reply]
Yep, you deploy _from_ a GitHub repository rather than deploying _to_ it.

If you need help creating a GitHub repository from a local git repository, here’s GitHub’s guide:

https://docs.github.com/en/migrations/importing-source-code/using-the-command-line-to-import-source-code/adding-locally-hosted-code-to-github

**Reply by christjohannes** [reply]
Thank you, this is working now

**Comment by shadoof** [comment]
And once the Framework project is in a Github repo, the static site can be built and deployed to Github Pages with a Github workflow like this one https://github.com/observablehq/framework/blob/main/.github/workflows/deploy.yml
